src/VertexManager.sol

```diff
@@ -365,24 +374,25 @@ contract VertexManager is IVertexManager, Initializable, UUPSUpgradeable, Ownabl
     /// @param id The ID of the pool to withdraw from.
     /// @param token The token to withdraw.
     /// @param amount The amount of token shares to withdraw.
-    function withdrawPerp(uint256 id, address token, uint256 amount) external whenWithdrawNotPaused nonReentrant {
+    function withdrawPerp(uint256 id, address token, uint256 amount)
+        external
+        payable
+        whenWithdrawNotPaused
+        nonReentrant
+    {
         // Fetch the pool storage.
         Pool storage pool = pools[id];
 
         // Check that the pool is perp.
         if (pool.poolType != PoolType.Perp) revert InvalidPool(id);
 
-        // Get the token storage.
```

Comment:
this was removed to reduce contract size, the checks are done in the `_withdraw` function